modules/bezug_solarworld/solarworld.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+from datetime import datetime, timezone
+import os
+import re
+import requests
+import sys
+import traceback
+
+solarworld_emanagerip = str(sys.argv[1])
+
+Debug         = int(os.environ.get('debug'))
+myPid         = str(os.getpid())
+
+def DebugLog(message):
+    local_time = datetime.now(timezone.utc).astimezone()
+    print(local_time.strftime(format = "%Y-%m-%d %H:%M:%S") + ": PID: "+ myPid +": " + message)
+
+if Debug >= 2:
+    DebugLog('Solarworld IP: ' + solarworld_emanagerip)
+
+# Auslesen eines Solarworl eManagers über die integrierte JSON-API
+emanagerantwort = requests.get('http://'+solarworld_emanagerip+'/rest/solarworld/lpvm/powerAndBatteryData', timeout=5).json()
+try:
+    em_in_watt = emanagerantwort["PowerIn"]
+except:
+    traceback.print_exc()
+    exit(1)
+try:
+    em_out_watt = emanagerantwort["PowerOut"]
+except:
+    traceback.print_exc()
+    exit(1)
+
+# Bezug ist entweder -Out oder In; bei Einspeisung ist 'em_in_watt' immer 0
+# use printf zum runden, LC_ALL=C wegen Dezimalpunkt
+bezug_watt = int(em_in_watt - em_out_watt)
+
+# wenn eManager aus bzw. keine Antwort ersetze leeren Wert durch eine 0
+ra = '^-?[0-9]+$'
+
+if re.search(bezug_watt, ra) == None:
+    bezug_watt = "0"
```

Comment:
Hier wieder ein garantierter `TypeError`. Selbst wenn es string wäre: `bezug_watt` ist auf jeden Fall ein `int` und der Regex würde damit garantiert IMMER matchen und damit die Zeile `bezug_watt = "0"` niemals zum Tragen kommen.

Hier sind wir aber endlich im Vorteil, weil der ursprüngliche Autor (danke, Kevin) so freundlich war uns die Intention in einem Kommentar zu hinterlegen. Wenn der eManager aus ist oder nicht antwortet, dann wird bereits in Zeile 22 eine Exception fliegen und wir kommen garnicht bis hier. Der Code kann also gestrichen (oder in ein try/except um Zeile 22 falls das wirklich gebraucht wird).